pydropsonde/processor.py

```diff
@@ -1166,13 +1166,44 @@ def add_attributes_as_var(self):
         (and not lose information)
         """
         _prep_l3_ds = self._prep_l3_ds
+        attr_list = []
         for attr, value in self._prep_l3_ds.attrs.items():
             _prep_l3_ds[attr] = value
+            attr_list.append(attr)
 
         _prep_l3_ds.attrs.clear()
+
+        object.__setattr__(self, "attrs", attr_list)
         object.__setattr__(self, "_prep_l3_ds", _prep_l3_ds)
         return self
 
+    def rename_attr_vars(self):
+        attrs = self.attrs
+        ds = self._prep_l3_ds
+        for attr in self.attrs:
+            splt = attr.split("(")
+            var_name = splt[0][:-1]
+            try:
+                unit = splt[1][:-1]
+                attrs.append(var_name)
+                ds = ds.rename({attr: var_name})
+                ds[var_name] = ds[var_name].assign_attrs(units=unit)
+            except IndexError:
+                pass
+        ds = ds.assign(
+            dict(
+                launch_time=(
+                    "sonde_id",
+                    [ds.launch_time.astype(np.datetime64).values],
+                    {"time_zone": ds.launch_time.attrs["units"]},
+                )
+            )
+        )
+
+        object.__setattr__(self, "attrs", attrs)
+        object.__setattr__(self, "_prep_l3_ds", ds)
+        return self
```

Comment:
Is this function basically splitting `launch_time(UTC)` into the variable name `launch_time` and time zone UTC? I am struggling a bit with the generic way the function is written while it seems to only handle this one specific case (if I understand it correctly). Could it also be exchanged with a function `correct_launch_time_varname` or something similar?

Reply:
it does not only split launch_time. there are others in the form `<var_name>_(<unit>)` (if I remember correctly some aircraft data like `aircraft_altitude_(m)` or similar. launch time is just the only one that is manually changed since UTC is not really a unit but rather a time zone

Reply:
So true! Sorry, it makes perfectly sense to do it like that. Cleans up the dataset even more :)